File: docs/src/testers/file-storage/force_overwrite.py
# @sniptest filename=force_overwrite.py
# @sniptest show=15-19
from pathlib import Path

from notte_sdk import NotteClient as FixtureClient

fixture = FixtureClient()
with fixture.Session() as session:
    uploaded = session.storage.upload("file.pdf")
    try:
        session_id = session.session_id
        file_id = uploaded.id
        destination = Path("downloads/file.pdf")
        destination.parent.mkdir()
        destination.write_bytes(b"old contents")

        from notte_sdk import NotteClient

        client = NotteClient()
        storage = client.FileStorage(session_id)
        storage.download(file_id, local_dir="./downloads", force=True)

        assert destination.read_bytes() == Path("file.pdf").read_bytes()
        status = session.status()
    finally:
        session.storage.delete(uploaded.id)

File: docs/src/testers/file-storage/force_overwrite.ts
// @sniptest filename=force_overwrite.ts
// @sniptest show=15-16
import assert from 'node:assert/strict';
import { mkdir, readFile, writeFile } from 'node:fs/promises';
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fixtureStorage = client.FileStorage();
const status = await client.Session({ storage: fixtureStorage }).use(async session => {
  const uploaded = await fixtureStorage.upload('file.pdf');
  try {
    const sessionId = fixtureStorage.sessionId;
    const fileId = uploaded.id;
    await mkdir('downloads', { recursive: true });
    await writeFile('downloads/file.pdf', 'old contents');

    const storage = client.FileStorage(sessionId);
    await storage.download(fileId, './downloads', { force: true });

    assert.deepEqual(await readFile('downloads/file.pdf'), await readFile('file.pdf'));
    return await session.status();
  } finally {
    await fixtureStorage.delete(uploaded.id);
  }
});
export { status };
